### ISSUE TITLE
BlockServer and RCPTT: Deleting configs/synoptics

### ISSUE THREAD

**Issue by DominicOram** [ISISComputingGroup/IBEX#1436]
Between system tests the cleanIBEXServer script removes all rcptt configs and synoptics from the file system. However, the BlockServer keeps the whole config/synoptic file system under version control and should pull them back in after it starts up. Interestingly, this behaviour doesn't seem to currently be working so this ticket is to fix this behaviour and come up with a way for RCPTT to delete configs/synoptics properly between tests.

**Comment by DominicOram** [comment]
The BlockServer version control should be tested as part of the system tests. This could be done as part of this ticket.

**Comment by kjwoodsISIS** [comment]
Consider modifying the clean-up script instead.

**Comment by IsabellaRey** [comment]
We thought the best way would be for the clean-up script to handle deleting configs/synoptics and pushing the changes to git, as we don't want the blockserver to do this for us (we don't want the blockserver to know about system tests, and if the blockserver version control should break, it would break the system tests cleanup too)

**Comment by IsabellaRey** [comment]
This ticket went through various discussions, and we analysed different solutions, taking into account that contexts in rcptt can only run before tests, not after. I'm documenting here just to keep track of them somewhere.

* Have the cleanup script commit changes to Git when rcptt files are deleted. The problem with this is that when the blockserver will be fixed to actually push changes to git (rather than having only local commits) we will end up pushing rcptt test files too, which we don't want to do.
* Have the cleanup script change branch on the config repository

Neither of the solutions above cleans up after the last test. To solve this, we could add an empty test that always gets run last, so the cleanup would still be executed, but on developers machines we often run only one test at the time: the first solution would leave us with files pushed to git, the second would leave us with the config in a test branch. The developer could run a python script to restore everything in the correct state, but it becomes annoying having to do this every time.

* Rcptt tests could be wrapped in a try-finally block: the finally block is always executed, so it could call a cleanup function. However, we would have to remember to add this every time. If we forget we're back at having the system left in a potentially bad state.

* We thought of running test scripts as contexts, so that cleanup contexts could run after. However, when a context fails, the following contexts are not executed. In addition to this, any cleanup context cannot be marked as default, so again we would have to remember to add them every time.

* We could add rcptt_* to the .gitignore of the config repository. That sounds like a really neat solution. The problem is that the git library used by the ConfigVersionControl (used by the blockserver) by default forces adding files to git, not honouring the .gitignore. There is a flag to tell it not to force, but it's not used!! We checked the latest version of this library on github, and they still haven't fixed it

The solution we went with is:
* The rcptt tests don't handle git at all. The ConfigVersionControl is set to ignore any file that contains "rcptt_" in its path. Therefore test files don't get added to git in the first place. This is not ideal, because the ConfigVersionControl has knowledge about something of the system tests, but it means any spurious files are only kept locally and developers don't need to remember extra steps when writing/running system tests locally.
